File: task-manager-main/frontend/src/services/calendar.service.js
import api from './api.service';
import { API_ENDPOINTS } from '../config/api';

const calendarService = {
  async getDailyView(date) {
    const response = await api.get(API_ENDPOINTS.CALENDAR.DAILY, {
      params: { date },
    });
    return response.data.data;
  },

  async getWeeklyView(date) {
    const response = await api.get(API_ENDPOINTS.CALENDAR.WEEKLY, {
      params: { date },
    });
    return response.data.data;
  },

  async getMonthlyView(date) {
    const response = await api.get(API_ENDPOINTS.CALENDAR.MONTHLY, {
      params: { date },
    });
    return response.data.data;
  },
};

export default calendarService;

File: task-manager-main/src/main/java/com/taskmanager/service/CalendarService.java
package com.taskmanager.service;

import com.taskmanager.dto.calendar.CalendarViewDTO;
import com.taskmanager.entity.User;
import com.taskmanager.strategy.CalendarViewStrategy;
import com.taskmanager.strategy.DailyCalendarView;
import com.taskmanager.strategy.MonthlyCalendarView;
import com.taskmanager.strategy.WeeklyCalendarView;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.time.LocalDate;

@Service
@RequiredArgsConstructor
@Slf4j
public class CalendarService {

    private final DailyCalendarView dailyCalendarView;
    private final WeeklyCalendarView weeklyCalendarView;
    private final MonthlyCalendarView monthlyCalendarView;
    private final UserService userService;

    @Transactional(readOnly = true)
    public CalendarViewDTO getDailyView(LocalDate date) {
        User user = userService.getCurrentUser();
        return executeStrategy(dailyCalendarView, user.getId(), date);
    }

    @Transactional(readOnly = true)
    public CalendarViewDTO getWeeklyView(LocalDate date) {
        User user = userService.getCurrentUser();
        return executeStrategy(weeklyCalendarView, user.getId(), date);
    }

    @Transactional(readOnly = true)
    public CalendarViewDTO getMonthlyView(LocalDate date) {
        User user = userService.getCurrentUser();
        return executeStrategy(monthlyCalendarView, user.getId(), date);
    }

    private CalendarViewDTO executeStrategy(CalendarViewStrategy strategy, Long userId, LocalDate date) {
        log.debug("Generating {} view for user {} with reference date {}",
                strategy.getViewType(), userId, date);
        return strategy.generateView(userId, date);
    }
}
